guides/source/routing.md

```diff
@@ -693,38 +731,40 @@ NOTE: You cannot override defaults via query parameters - this is for security r
 
 ### Naming Routes
 
-You can specify a name for any route using the `:as` option:
+You can specify a name that will used by the `_path` and `_url` helpers for any route using the `:as` option:
 
 ```ruby
 get 'exit', to: 'sessions#destroy', as: :logout
 ```
 
-This will create `logout_path` and `logout_url` as named route helpers in your application. Calling `logout_path` will return `/exit`
+This will create `logout_path` and `logout_url` as the route helpers in your application. Calling `logout_path` will return `/exit`.
 
-You can also use this to override routing methods defined by resources by placing custom routes before the resource is defined, like this:
+You can also use `as` to override routing helper names defined by `resources` by placing a custom route definition *before* the resource is defined, like this:
 
 ```ruby
 get ':username', to: 'users#show', as: :user
 resources :users
 ```
 
-This will define a `user_path` method that will be available in controllers, helpers, and views that will go to a route such as `/bob`. Inside the `show` action of `UsersController`, `params[:username]` will contain the username for the user. Change `:username` in the route definition if you do not want your parameter name to be `:username`.
+This will define a `user_path` helper that will match `/:username` (e.g. `/jane`). Inside the `show` action of `UsersController`, `params[:username]` will contain the username for the user.
 
 ### HTTP Verb Constraints
 
-In general, you should use the [`get`][], [`post`][], [`put`][], [`patch`][], and [`delete`][] methods to constrain a route to a particular verb. You can use the [`match`][] method with the `:via` option to match multiple verbs at once:
+In general, you should use the [`get`][], [`post`][], [`put`][], [`patch`][], and [`delete`][] methods to constrain a route to a particular verb. There is a [`match`][] method that you could use with the `:via` option to match multiple verbs at once:
 
 ```ruby
 match 'photos', to: 'photos#show', via: [:get, :post]
 ```
 
+The above route matches GET and POST requests to the `show` action of the `PhotosController`.
+
 You can match all verbs to a particular route using `via: :all`:
 
 ```ruby
 match 'photos', to: 'photos#show', via: :all
 ```
 
-NOTE: Routing both `GET` and `POST` requests to a single action has security implications. In general, you should avoid routing all verbs to an action unless you have a good reason to.
+NOTE: Routing both `GET` and `POST` requests to a single action has security implications. In general, you should avoid routing all verbs to an action unless you have a good reason.
```

Comment:
Is the next NOTE one of the "security implications" of this? If so, could it be just a single NOTE?

Reply:
Sure!